OneCommunityGlobal/HGNRest
# Vishala Leaderboard and Team Member Tasks Api Fix

# Description
This PR is about the Leaderboard api fix and Team Member Tasks api fix. 

## Related PRS (if any):
To test this backend PR you need to checkout the development frontend branch.

## Main changes explained:
- The existing query for leaderboard and teamMemberTasks failed due to heavy load on the aggregation pipeline. Re wrote the queries without aggregation.

## How to test:
1. check into current branch
2. do `npm run build` and `npm run start` to run this PR locally
3. Open development branch on front end
4. Test if the leaderboard data is properly being displayed based on user role 
5. Test if the tasks of team members are being displayed properly based on the user role
6. Owner and Core Team can view all active users
7. Admin can view all active users except Owner and Core Team
8. Other can view their team members

## Screenshots or videos of changes:

[Untitled_ Jan 03,2024 2_37 PM.webm](https://github.com/OneCommunityGlobal/HGNRest/assets/65274029/36a48724-f766-4fda-bb1d-2fc2fbff3a1a)

<img width="407" alt="image" src="https://github.com/OneCommunityGlobal/HGNRest/assets/65274029/bf09b412-665f-488c-81e5-c7f3f39179b4">
<img width="530" alt="image" src="https://github.com/OneCommunityGlobal/HGNRest/assets/65274029/ecad3638-01e9-4c43-9355-822fbb91c806">
